app/forms/openid_connect_authorize_form.rb

```diff
@@ -56,6 +56,13 @@ def submit(user, rails_session_id)
     FormResponse.new(success: success, errors: errors.messages, extra: extra_analytics_attributes)
   end
 
+  # Similar to submit, but only helps generate errors
+  def check_submit
```

Comment:
Can I push the change that I'm thinking of? I verified that all tests pass with my proposed changes. Maybe that will better explain what I have in mind. Basically, I don't think we need to be validating the request twice. By validating earlier in the process, we know that once it reaches the `index` action, it is valid. If it is not valid, it will never reach the `index` action.

This means we don't need a new `check_submit` method. We only need to modify the existing `submit` method to only validate and return the FormResponse, and link the identity in the `index` action once the user has signed in.

This is consistent with what we do with SAML: First we validate the SAML request, and if it's valid, then we proceed to the `auth` action, and once the user is fully authenticated, we link the identity: https://github.com/18F/identity-idp/blob/master/app/controllers/saml_idp_controller.rb#L41-L50. 

This also reminds me we need to update the OIDC controller (in a separate PR) to match the latest changes in SAML for the scenario where a user needs to finish verifying their profile (when they are waiting to confirm their mailing address for example). Eventually, it would be great if we could abstract out these controllers so we don't have to remember to keep them in sync.